src/main/java/gov/cabinetoffice/gap/applybackend/service/GrantApplicantOrganisationProfileService.java

```diff
@@ -37,4 +38,19 @@ public GrantApplicantOrganisationProfile createOrganisation(String applicantId,
         grantApplicantService.saveApplicant(applicant);
         return savedProfile;
     }
+
+    public Boolean isOrganisationComplete(final String sub) {
```

Comment:
This ones a bit sore on the eye. Thinking we can be clever here and use a generic to check if any of the fields are null or empty.  What do you think?

` public Boolean isOrganisationComplete(final String sub) {
        final GrantApplicant applicant = grantApplicantService.getApplicantById(sub);
        final GrantApplicantOrganisationProfile profile = applicant.getOrganisationProfile();

        return profile != null && allFieldsNotNullOrEmpty(profile);
    }

    private <T> boolean allFieldsNotNullOrEmpty(T object) {
        Class<?> clazz = object.getClass();
        Field[] fields = clazz.getDeclaredFields();

        for (Field field : fields) {

            try {
                Object value = field.get(object);
                if (value == null || value.toString().isEmpty()) {
                    return false;
                }
            } catch (IllegalAccessException e) {
                e.printStackTrace();
            }
        }

        return true;
    }`

Reply:
Yeah originally this was in javascript, and went down a similar thought path. But nah not every field is actually required, and if you filter out the ones that arent it becomes just as convoluted

Reply:
Resolving after speaking with Dom - it appears not every field is required